plugins/woocommerce/tests/php/includes/wc-stock-functions-tests.php

```diff
@@ -432,4 +432,28 @@ public function test_wc_update_product_stock_no_stock_action() {
 
 		remove_action( 'woocommerce_no_stock', $callback );
 	}
+
+	/**
+	 * @testdox The wc_trigger_stock_change_actions function should only trigger actions if the product is set
+	 *          to manage stock.
+	 */
+	public function test_wc_trigger_stock_change_actions_bails_early_for_unmanaged_stock() {
+		$action_fired = false;
+		$callback     = function () use ( &$action_fired ) {
+			$action_fired = true;
+		};
+		add_action( 'woocommerce_no_stock', $callback );
+
+		$product = WC_Helper_Product::create_simple_product();
+
+		$this->assertFalse( $action_fired );
```

Comment:
Product creation takes care of firing the action, but perhaps we want to be more   explicit here, in case if the behaviour ever changes.
```suggestion
        $product->set_stock_quantity( 0 );
		$this->assertFalse( $action_fired );
```